### ISSUE TITLE
Script is executed multiple times under different filenames/scopes with buggy output

### ISSUE THREAD

**Issue by dogancelik** [greasemonkey/greasemonkey#2308]
Hello, I found this strange bug, where **a single user script** is executed multiple times under **different filenames** (from a site's own script, not a user script).

```js
// ==UserScript==
// @name        Twitch Test
// @namespace   test
// @include     http*://www.twitch.tv/*
// @version     1.0.0
// @grant       none
// ==/UserScript==

(function() {

setInterval(function () {
  console.log('test', document.getElementById('nav_personal'));
}, 1000);

})()
```

and here's the log from a Twitch.TV page http://www.twitch.tv/example

```
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
build: local global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
test null twitch_test.user.js:12:3
test <div class="nav_section" id="nav_personal"> global-87e0128c7e9b2be432c0f5aaf1e0947d.js:30:10014
test null twitch_test.user.js:12:3
```

You get this huge output (**37 lines**) under few seconds (**max: 3 seconds**).

* This script gets executed under a different script named: `global-87e0128c7e9b2be432c0f5aaf1e0947d.js`
* This is the script's URL: `http://www-cdn.jtvnw.net/assets/global-87e0128c7e9b2be432c0f5aaf1e0947d.js`
* Our original file `twitch_test.user.js`, gets ***null*** from *getElementById* but `global-87e0128c7e9b2be432c0f5aaf1e0947d.js` from **does not**.

I  tested this script under different conditions:

* Firefox 41 (Latest Stable) with only Greasemonkey 3.5 installed
* Firefox 41 with only Greasemonkey 3.0 installed
* Firefox 41 with only Greasemonkey 2.3 installed
* Firefox 43 (Latest Aurora) with only Greasemonkey 3.5 installed
* Firefox 43 with only Greasemonkey 3.5 beta installed
* Firefox 43 with only Greasemonkey 3.3 installed

**Comment by arantius** [comment]
And if you put `@noframes` in the meta?

**Comment by dogancelik** [comment]
@arantius No change

**Comment by janekptacijarabaci** [comment]
This isn't the bug.

I trying make a new profile of Firefox - GM is not installed:
![1_](https://cloud.githubusercontent.com/assets/2373486/10722676/fcbca64e-7bb2-11e5-979e-b70b10664bf4.png)

http://www-cdn.jtvnw.net/assets/global-87e0128c7e9b2be432c0f5aaf1e0947d.js:
![2](https://cloud.githubusercontent.com/assets/2373486/10722662/c31db8ba-7bb2-11e5-9428-777f66b1dd0a.png)

**Comment by dogancelik** [comment]
I feel like that script is overriding console.log, which is a bad practice...

**Comment by dogancelik** [comment]
I made this script to figure out how many frames the page has, its frames length changes frequently from 8 (start) to 6 (end).

I think I have figured this out.

* if getElementById() is null, it is executed from a child frame
* We always see Element in global-*.js because there is only one Top frame and that top frame has `#nav_personal` only and also console.log is overridden here.

I think we can close this, sorry for the troubles, still a bad move on their end for polluting browser functions.
